sdk/tables/azure-data-tables/tests/test_challenge_auth_async.py

```diff
@@ -7,15 +7,74 @@
 from unittest.mock import Mock
 
 from azure.core.credentials import AccessToken
+from azure.core.exceptions import ClientAuthenticationError
 from azure.core.pipeline import AsyncPipeline
 from azure.core.pipeline.transport import HttpRequest as PipelineTransportHttpRequest
-from azure.core.rest import HttpRequest as RestHttpRequest
+from azure.data.tables.aio import TableServiceClient
 from azure.data.tables.aio._authentication_async import AsyncBearerTokenChallengePolicy
 import pytest
 
+from async_preparers import tables_decorator_async
+from devtools_testutils import AzureRecordedTestCase, is_live
+from devtools_testutils.aio import recorded_by_proxy_async
+from _shared.asynctestcase import AsyncTableTestCase
+
 pytestmark = pytest.mark.asyncio
 
-HTTP_REQUESTS = [PipelineTransportHttpRequest, RestHttpRequest]
+# Try to use azure.core.rest request if azure-core version supports it -- fall back to basic request
+try:
+    from azure.core.rest import HttpRequest as RestHttpRequest
+    HTTP_REQUESTS = [PipelineTransportHttpRequest, RestHttpRequest]
+except ModuleNotFoundError:
+    HTTP_REQUESTS = [PipelineTransportHttpRequest]
+
+
+class TestTableChallengeAuthAsync(AzureRecordedTestCase, AsyncTableTestCase):
+    @tables_decorator_async
+    @recorded_by_proxy_async
+    async def test_challenge_auth_supported_version(self, tables_storage_account_name):
```

Comment:
These tests will be skipped with the CI live runs?

Reply:
Yep, that's correct. The default settings for `New-TestResources` service principals don't enable multitenant authentication, so there's no easy way to automate these tests in live mode (that I know of)

Reply:
Perfect - thanks!